### PR TITLE
Change flag name and simplify redirect logic #79

### PR DESCRIPTION
**Pull request by pdbreen**

### CODE REVIEW

**Comment by drewroberts** [comment]
The logic for the Page's path looks perfect to me:

https://github.com/drewroberts/blog/blob/495c222a0ae424a7e6d1fed1a4aee6e10d4fa535/src/Models/Page.php#L140-L157

File: src/Models/Page.php [hunk 1]
```diff
@@ -139,27 +139,23 @@ public function layout()
...
+        return empty($path) ? '/' : implode('/', array_reverse($path));
     }
 
     public function getIsOnlyRootLocationAttribute(): bool
     {
-        return ($this->location_based && $this->is_root &&
-            static::query()->whereNull('parent_id')->where('location_based', true)->count() === 1);
+        return ($this->is_location && $this->is_root &&
+            static::query()->whereNull('parent_id')->where('is_location', true)->count() === 1);
     }
```
Comment (drewroberts) [inline src/Models/Page.php:159]: OK, that's even cleaner! Thank you

**Review by drewroberts — APPROVED** [review]
Thank you! 🏆

File: src/Http/Controllers/PageController.php [hunk 2]
```diff
@@ -13,24 +13,11 @@ class PageController extends BaseController
...
-            // If location based child is only child, redirect!!!
-            if ($childPage->location_based && $childPage->is_only_child) {
-                return redirect(url($childPage->path));
-            }
-        } else {
-            // If there is only  location based root page, redirect home!
-            if ($page->is_only_root_location) {
-                return redirect(url($page->path));
-            }
+        $leafPage = $grandChildPage ?: ($childPage ?: $page);
```
Comment (jasonmccreary) [inline src/Http/Controllers/PageController.php:16]: I see this logic in a few spots. It's a bit hard to parse the compound ternary and sooner or later a dev is going to forget those parenthesis which will alter the behavior.

Wonder if wrapping in a simple helper would be... helpful: `determine_leaf_page($grandChildPage, $childPage, $page)`

**Review by jasonmccreary — COMMENTED** [review]
Minor suggestion.
